zproject/backends.py

```diff
@@ -243,6 +243,14 @@ class ZulipAuthMixin:
     from memcached before checking the database (avoiding a database
     query in most cases).
     """
+    _logger = None
+    name: str  # this is to avoid mypy error no attribute "name" for below code. https://github.com/python/mypy/issues/5868#issuecomment-437690894
+
+    @property
+    def logger(self) -> logging.Logger:
+        if self._logger is None:
+            self._logger = logging.getLogger(f"zulip.auth.{self.name}")
```

Comment:
I don't think all backends have `name` attribute, let's just use `self.__class__.__name__` here instead perhaps?

Reply:
I think it'd be nicer to use the shorter `name` attributes.  Maybe worth migrating to have that be the case; it might also allow us to clean up things like this:
```
AUTH_BACKEND_NAME_MAP: Dict[str, Any] = {                                                            
    'Dev': DevAuthBackend,                                                                           
    'Email': EmailAuthBackend,                                                                       
    'LDAP': ZulipLDAPAuthBackend,                                                                    
}                                                                                                    
```

(Those should be the `name` attributes for those backends, right?).

Reply:
@mateuszmandera from the code I see that only backends like github, google, gitlab, saml. LDAP have logging calls. LDAP seems to have a different logger. So, I haven't touched anything related to LDAP. And those backends that have logging calls have `name` too.

@timabbott yes, but we usually have `name` in lowercase and the normal one with mix of upper and lowercase letters is named `auth_backend_name` which is used as display name in login and sign up pages. I do not see any of those having logging calls except `LDAP`, so I'm not sure if it's useful to have a name attribute for `dev` and `Email` auth backends.

Reply:
Hmm, it might make sense to make `name` and `auth_backend_name` be class attributes on all backends, not just the `@external_auth_method` ones, even if only for consistency. And there's no reason to expect we won't ever want to add logging in those other backends :thinking:

Reply:
Okay, added a commit adding `name` and `auth_backend_name` for the those three too.